src/main/drivers/system_stm32h7xx.c

```diff
@@ -0,0 +1,357 @@
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
+    GPIO_InitStruct.Alternate = GPIO_AF0_MCO;
+
+    HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+    GPIO_InitStruct.Pin = GPIO_PIN_9;
+    GPIO_InitStruct.Mode = GPIO_MODE_AF_PP;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
+    GPIO_InitStruct.Alternate = GPIO_AF0_MCO;
+
+    HAL_GPIO_Init(GPIOC, &GPIO_InitStruct);
+}
+
+bool isMPUSoftReset(void)
+{
+    if (cachedRccCsrValue & RCC_RSR_SFTRSTF)
+        return true;
+    else
+        return false;
+}
+
+void systemInit(void)
+{
+#ifdef USE_ITCM_RAM
+    //  Mark ITCM-RAM as read-only
+    HAL_MPU_Disable();
+
+    // "For Cortex®-M7, TCMs memories always behave as Non-cacheable, Non-shared normal memories, irrespectiveof the memory type attributes defined in the MPU for a memory region containing addresses held in the TCM"
+    // See AN4838
+
+    MPU_Region_InitTypeDef MPU_InitStruct;
+    MPU_InitStruct.Enable = MPU_REGION_ENABLE;
+    MPU_InitStruct.BaseAddress = 0x00000000;
+    MPU_InitStruct.Size = MPU_REGION_SIZE_64KB;
+    MPU_InitStruct.AccessPermission = MPU_REGION_PRIV_RO_URO;
+    MPU_InitStruct.IsBufferable = MPU_ACCESS_BUFFERABLE;
+    MPU_InitStruct.IsCacheable = MPU_ACCESS_NOT_CACHEABLE;
+    MPU_InitStruct.IsShareable = MPU_ACCESS_NOT_SHAREABLE;
+    MPU_InitStruct.Number = MPU_REGION_NUMBER0;
+    MPU_InitStruct.TypeExtField = MPU_TEX_LEVEL0;
+    MPU_InitStruct.SubRegionDisable = 0x00;
+    MPU_InitStruct.DisableExec = MPU_INSTRUCTION_ACCESS_ENABLE;
+    HAL_MPU_ConfigRegion(&MPU_InitStruct);
+
+    HAL_MPU_Enable(MPU_PRIVILEGED_DEFAULT);
+#endif
+
+
+    // Configure NVIC preempt/priority groups
+    HAL_NVIC_SetPriorityGrouping(NVIC_PRIORITY_GROUPING);
+
+    // cache RCC->RSR value to use it in isMPUSoftReset() and others
+    cachedRccCsrValue = RCC->RSR;
+
+    /* Accounts for OP Bootloader, set the Vector Table base address as specified in .ld file */
+    //extern void *isr_vector_table_base;
+    //NVIC_SetVectorTable((uint32_t)&isr_vector_table_base, 0x0);
+    //__HAL_RCC_USB_OTG_FS_CLK_DISABLE;
+
+    //RCC_ClearFlag();
+
+    enablePeripherialClocks();
+
+    enableGPIOPowerUsageAndNoiseReductions();
+
+#ifdef USE_MCO_OUTPUTS
+    configureMasterClockOutputs();
+#endif
+
+    // Init cycle counter
+    cycleCounterInit();
+
+    // SysTick is updated whenever HAL_RCC_ClockConfig is called.
+}
+
+void systemReset(void)
+{
+#if 0
+#ifdef USE_GYRO
+    if (mpuResetFn) {
+        mpuResetFn();
+    }
+#endif
+#endif
+
+    SCB_DisableDCache();
+    SCB_DisableICache();
+
+    __disable_irq();
+    NVIC_SystemReset();
+}
+
+void forcedSystemResetWithoutDisablingCaches(void)
+{
+    persistentObjectWrite(PERSISTENT_OBJECT_RESET_REASON, RESET_FORCED);
+
+    __disable_irq();
+    NVIC_SystemReset();
+}
+
+void systemResetToBootloader(void)
+{
+#if 0
+#ifdef USE_GYRO
+    if (mpuResetFn) {
+        mpuResetFn();
+    }
+#endif
+#endif
+#ifdef EXST
```

Comment:
Should this be `USE_EXST`?

Reply:
I have discussed this with @hydra. The `EXST` was used following the `OPBL` legacy, but we both are not fussed. The `USE_EXST` was considered also, but to me, it's a bit awkward as `EXST` is not a feature but how firmware is built.  Thoughts?

Reply:
To me, `USE_` defines are just binary type modifiers to the build process. (And they have the beneficial side-effect of making it much easier to grep for them in the source.)

Reply:
Hmm... I think the `USE_EXST` makes me uncomfortable because `EXST` will going to appear in `STM32H7.mk`.

In any case, the `EXST` code was leaked and should not appear at this phase, so I'll just drop them from this PR. We can resume/restart this discussion when `EXST` build is actually implemented.

Reply:
marking as resolved as EXST has gone from this PR now.